HF prep: generate a keypair restructure, plus node operator metadata and light edits (#605)
o1-labs/docs2: docs/node-operators/snark-workers.mdx

---
title: SNARK Workers
hide_title: true
---

# SNARK Workers

> While most protocols have just one primary group of node operators (often called miners, validators, or block producers), Mina has a second group — the **snark worker**.

Author: ```suggestion
While most protocols have just one primary group of node operators (often called miners, validators, or block producers), Mina has a second group — the **SNARK worker**.
```

SNARK workers are integral to the Mina network's health because these nodes are responsible for snarking, or producing SNARK proofs, of transactions in the network. By producing these proofs, snark workers help maintain the succinctness of the Mina blockchain.

> In this document, we'll briefly cover why snark workers are needed, how the economic incentives align, and provide operational details of performing snark work. Feel free to click through to any of the sections above that are most relevant to your needs.
> 
> Note: This document primarily targets node operators, and will not cover the theory of zk-SNARKs. Deep knowledge of SNARKs is not required to read this section, but it will be helpful to know roughly how SNARKs work and what they are useful for. If this is not the case, please check out [this primer on SNARKs](https://minaprotocol.com/blog/what-are-zk-snarks) first.

Author: ```suggestion
Read on to learn why SNARK workers are needed, how the economic incentives align, and operational details of performing SNARK work. Feel free to click through to any of the sections that are most relevant to your needs.

Note: The theory of zk-SNARKs is not covered. Deep knowledge of SNARKs is not required to read this section, but it is helpful to understand in general how SNARKs work and what they are useful for. To learn more, check out this
 [What are zk-SNARKs?](https://minaprotocol.com/blog/what-are-zk-snarks) primer first.
```
